src/main/java/org/truffleruby/language/arguments/RubyArguments.java

```diff
@@ -115,31 +118,37 @@
         return newArgs;
     }
 
-    public static boolean assertFrameArguments(Object[] frameArguments) {
-        Object callerData = getCallerData(frameArguments);
-        InternalMethod method = getMethod(frameArguments);
-        DeclarationContext declarationContext = getDeclarationContext(frameArguments);
-        Object self = getSelf(frameArguments);
-        Object block = getBlock(frameArguments);
-        Object[] arguments = getArguments(frameArguments);
-
-        assert method != null;
-        assert declarationContext != null;
-        assert self != null;
+    public static boolean assertArguments(Object[] arguments) {
```

Comment:
Should still be named `assertFrameArguments` for clarity it's not just user arguments.